python/cudf/cudf/core/reshape.py

```diff
@@ -38,36 +40,66 @@ def _align_objs(objs, how="outer"):
 
         index = objs[0].index
         name = index.name
-        if how == "inner" or isinstance(index, cudf.MultiIndex):
-            for obj in objs[1:]:
-                index = (
-                    cudf.DataFrame(index=obj.index)
-                    .join(cudf.DataFrame(index=index), how=how)
-                    .index
-                )
-                index.name = name
-            return [obj.reindex(index) for obj in objs], False
 
-        else:
-            all_index_objs = [obj.index for obj in objs]
-            appended_index = all_index_objs[0].append(all_index_objs[1:])
-            df = cudf.DataFrame(
-                {
-                    "idxs": appended_index,
-                    "order": cudf.core.column.arange(
-                        start=0, stop=len(appended_index)
-                    ),
-                }
+        if how == "inner":
+            final_index = _get_combined_index(
+                [obj.index for obj in objs], intersect=True, sort=sort
             )
-            df = df.drop_duplicates(subset=["idxs"]).sort_values(
-                by=["order"], ascending=True
+        else:
+            final_index = _get_combined_index(
+                [obj.index for obj in objs], intersect=False, sort=sort
             )
```

Comment:
I can't make a suggestion here because there are deleted lines interspersed with the new code, but this can just be:
```
final_index = _get_combined_index(
    [obj.index for obj in objs], intersect=(how == "inner"), sort=sort
    )
)
```